content/connectors/examples/github.md

```diff
@@ -0,0 +1,379 @@
+  "license": {
+    "key": "apache-2.0",
+    "name": "Apache License 2.0",
+    "spdx_id": "Apache-2.0",
+    "url": "https://api.github.com/licenses/apache-2.0",
+    "node_id": "MDc6TGljZW5zZTI="
+  },
+  "allow_forking": true,
+  "is_template": false,
+  "topics": [
+    "cloud-native",
+    "rust",
+    "rust-lang",
+    "streaming"
+  ],
+  "visibility": "public",
+  "forks": 50,
+  "open_issues": 103,
+  "watchers": 790,
+  "default_branch": "master",
+  "temp_clone_token": null,
+  "organization": {
+    "login": "infinyon",
+    "id": 52172389,
+    "node_id": "MDEyOk9yZ2FuaXphdGlvbjUyMTcyMzg5",
+    "avatar_url": "https://avatars.githubusercontent.com/u/52172389?v=4",
+    "gravatar_id": "",
+    "url": "https://api.github.com/users/infinyon",
+    "html_url": "https://github.com/infinyon",
+    "followers_url": "https://api.github.com/users/infinyon/followers",
+    "following_url": "https://api.github.com/users/infinyon/following{/other_user}",
+    "gists_url": "https://api.github.com/users/infinyon/gists{/gist_id}",
+    "starred_url": "https://api.github.com/users/infinyon/starred{/owner}{/repo}",
+    "subscriptions_url": "https://api.github.com/users/infinyon/subscriptions",
+    "organizations_url": "https://api.github.com/users/infinyon/orgs",
+    "repos_url": "https://api.github.com/users/infinyon/repos",
+    "events_url": "https://api.github.com/users/infinyon/events{/privacy}",
+    "received_events_url": "https://api.github.com/users/infinyon/received_events",
+    "type": "Organization",
+    "site_admin": false
+  },
+  "network_count": 50,
+  "subscribers_count": 21
+}
+```
+
+As we can see, the data we want is in here (`stargazers_count` and `forks_count`),
+but it's buried under the slew of other fields that we don't need for our use-case. If we kept a
+long-lived stream of this data running, we'd be using hundreds of times the bandwidth necessary.
+
+Let's see how we can use a custom SmartModule to select just these three fields we're interested
+in to create a lean and lightweight stream that meets our needs.
+
+### Using a SmartModule to shape GitHub data
+
+Now we're going to take a look at how a SmartModule can help us slim down our HTTP responses.
+We'll be using [this example GitHub SmartModule][2], which we'll learn more about in this section.
+If you'd like to create your own SmartModule from scratch, see [the docs on SmartModule Maps][3].
+
+Since we'll be using a pre-written SmartModule, the first thing we need to do is clone the repository:
+
+%copy first-line%
+```bash
+$ git clone https://github.com/infinyon/fluvio-smartmodule-examples && cd fluvio-smartmodule-examples
+```
+
+Then, let's take a look at the GitHub SmartModule example at `github-stars/src/lib.rs`.
+
+```rust
+use fluvio_smartmodule::{smartmodule, Record, RecordData, Result};
+use serde_json::{json, Value as JsonValue};
+
+#[smartmodule(map)]
+fn shaper(record: &Record) -> Result<(Option<RecordData>, RecordData)> {
```

Comment:
can we also add a `timestamp` ? 
```
    let shaped = json!({
        "time": "2021-12-02 21:10:23", (API to compute this)
        "stars": repo["stargazers_count"],
        "forks": repo["forks_count"],
    });
```

Reply:
I just tried it and I don't think that WASM modules have access to the system clock, so we can't detect the timestamp of when we are processing this entry. However, the input object has an `updated_at` field that has a timestamp of when the repository was last "updated". If you want, we could include that, but I don't know if it's what we really want, it won't be the time when we last got a star or fork, it will be the last time an "update" happened to the repo, and I'm not sure exactly what GitHub considers an "update".

Reply:
sure, let's do that.